File: sdk/runanywhere-android/src/main/java/com/runanywhere/sdk/models/HardwareConfiguration.kt
package com.runanywhere.sdk.models

/**
 * Hardware configuration for framework adapters
 */
data class HardwareConfiguration(
    var primaryAccelerator: HardwareAcceleration = HardwareAcceleration.AUTO,
    var fallbackAccelerator: HardwareAcceleration? = HardwareAcceleration.CPU,
    var memoryMode: MemoryMode = MemoryMode.BALANCED,
    var threadCount: Int = Runtime.getRuntime().availableProcessors(),
    var useQuantization: Boolean = false,
    var quantizationBits: Int = 8
) {
    enum class MemoryMode {
        CONSERVATIVE,
        BALANCED,
        AGGRESSIVE
    }
} 

File: sdk/runanywhere-swift/Sources/RunAnywhere/Capabilities/DeviceCapability/Models/HardwareConfiguration.swift
import Foundation

/// Hardware configuration for framework adapters
public struct HardwareConfiguration {
    public var primaryAccelerator: HardwareAcceleration = .auto
    public var fallbackAccelerator: HardwareAcceleration? = .cpu
    public var memoryMode: MemoryMode = .balanced
    public var threadCount: Int = ProcessInfo.processInfo.processorCount
    public var useQuantization: Bool = false
    public var quantizationBits: Int = 8

    public enum MemoryMode {
        case conservative
        case balanced
        case aggressive
    }

    public init(
        primaryAccelerator: HardwareAcceleration = .auto,
        fallbackAccelerator: HardwareAcceleration? = .cpu,
        memoryMode: MemoryMode = .balanced,
        threadCount: Int = ProcessInfo.processInfo.processorCount,
        useQuantization: Bool = false,
        quantizationBits: Int = 8
    ) {
        self.primaryAccelerator = primaryAccelerator
        self.fallbackAccelerator = fallbackAccelerator
        self.memoryMode = memoryMode
        self.threadCount = threadCount
        self.useQuantization = useQuantization
        self.quantizationBits = quantizationBits
    }
}
